bin/npg_publish_tree.pl

```diff
@@ -188,46 +188,32 @@ sub handler {
 my @files = grep { -f } $publisher->list_directory($source_directory,
                                                    recurse => 1);
 
-my $mlwh_json_cb = sub {
-  # DataObject, path to file folder, collection file
-  my ($obj, $collection, $file) = @_;
-  if (defined $mlwh_json_filename) { 
+sub write_json {
+  my ($json_filename, $irods_collection) = @_;
+  if (defined $json_filename && (not $json_filename =~ qr/^\s*$/sxm)
+      && defined $irods_collection && (not $irods_collection =~ qr/^\s*$/sxm)) {
     Readonly::Scalar my $JSON_FILE_VERSION => '1.0';
-    my $mlwh_hash = {
-      irods_root_collection    => $collection,
-      irods_data_relative_path => $file
-    };
-    my ($json_fh, $json_hash);
-    if (-e $mlwh_json_filename) {
-      open $json_fh, '+<:encoding(UTF-8)', $mlwh_json_filename or
-        self->logcroak(q[could not open ml warehouse json file] .
-        qq[$mlwh_json_filename]);
-      $json_hash = decode_json <$json_fh>;
-    }
-    else {
-      open $json_fh, '>:encoding(UTF-8)', $mlwh_json_filename or
-        self->logcroak(q[could not open ml warehouse json file] .
-        qq[$mlwh_json_filename]);
-      $json_hash = {
-        version  => $JSON_FILE_VERSION,
-        irods_top_collection_folder => $irods_tmp_coll,
-        products => [],
-      };
-    }
-    push @{$json_hash->{products}}, $mlwh_hash;
-
-    seek $json_fh, 0, 0;
 
+    my ($json_fh, $json_hash);
+    open $json_fh, '>:encoding(UTF-8)', $json_filename or
+      self->logcroak(q[could not open ml warehouse json file] .
+      qq[$json_filename]);
+    $json_hash = {
+      version  => $JSON_FILE_VERSION,
+      irods_collection => $irods_collection
```

Comment:
You're declaring the same JSON file version as the in Illumina run publisher, but using a different JSON property for the root collection value. These should be consistent.

Reply:
Should I change the json version? In this case we do not have the product list.

Reply:
Is there a reason not to follow the same structure?

Reply:
If there is, I would say a different version number would be useful to allow scripts using them (e.g. the loader) to differentiate and treat them differently

Reply:
This json file should contain only the irods_collection value. No products appended.

Reply:
I don't think I understand the purpose of this pr. If there is no information other than a collection, what is the point of it being json encoded? You could just write out a file with the collection as plain text.

Reply:
My next question would be: how should we handle the file version? If this raises some problems, probably we could follow Michael advice and write to a plain text file. any thoughts?